### PR TITLE
feature-branch-backend-continuation into development

### PR DESCRIPTION
**Pull request by julianertle**
Implemented all needed functionalities for the backend

### CODE REVIEW

**Comment by FelixAlexK** [comment]
@julianertle, you have to solve conflicts first

File: app/src/main/java/com/example/maraudersmap/LocationXTO.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.example.maraudersmap
+
+class LocationXTO {
```
Comment (FelixAlexK) [inline app/src/main/java/com/example/maraudersmap/LocationXTO.kt:3]: Are [Data Classes](https://www.baeldung.com/kotlin/data-classes) more useful here?

File: app/src/main/java/com/example/maraudersmap/UserXTO.kt [hunk 2]
```diff
@@ -4,4 +4,5 @@ class UserXTO {
     var username: String? = null
```
Comment (FelixAlexK) [inline app/src/main/java/com/example/maraudersmap/UserXTO.kt:4]: Are [Data Classes](https://www.baeldung.com/kotlin/data-classes) more useful here?

File: app/src/main/java/com/example/maraudersmap/ServerCommunicator.kt [hunk 3]
```diff
@@ -33,22 +36,69 @@ class ServerCommunicator {
...
 
     /**
-     * Posts a request to the server and returns the response
+     * Sends a posts request to the server and returns the response
      *
      * @param url Server address that needs to be communicated with
      * @param xmlObject Xml object that needs to be converted to an xml String. It contains the information that needs to be posted to the server
      * @return Response of the request
      */
     suspend fun <T> postRequest(url: String?, xmlObject: T): Response {
```
Comment (FelixAlexK) [inline app/src/main/java/com/example/maraudersmap/ServerCommunicator.kt:45]: Add try/catch blocks

File: app/src/main/java/com/example/maraudersmap/ServerCommunicator.kt [hunk 4]
```diff
@@ -33,22 +36,69 @@ class ServerCommunicator {
...
+    }
+
+    /**
+     * Sends a put request to the server and returns the response
+     *
+     * @param url Server address that needs to be communicated with
+     * @param xmlObject Xml object that needs to be converted to an xml String. It contains the information that needs to be posted to the server
+     * @return Response of the request
+     */
+    suspend fun <T> putRequest(url: String?, xmlObject: T): Response{
```
Comment (FelixAlexK) [inline app/src/main/java/com/example/maraudersmap/ServerCommunicator.kt:69]: Add try/catch blocks

**Review by FelixAlexK — CHANGES_REQUESTED** [review]
Overall it looks good but minor changes are needed

**Comment by julianertle** [comment]
@FelixAlexK conflicts resolved

**Comment by FelixAlexK** [comment]
> @julianertle 👍

File: app/src/main/java/com/example/maraudersmap/ServerCommunicator.kt (hunk 4 above)
Comment (julianertle) [reply app/src/main/java/com/example/maraudersmap/ServerCommunicator.kt:69]: Will do that in a new branch, functionality is implemented (under the condition the server is reachable) and it allows to be worked with

**Review by FelixAlexK — APPROVED** [review]
Looks good

**Pull request merged**